src/protocol/jdbc/org/apache/jmeter/protocol/jdbc/AbstractJDBCTestElement.java

```diff
@@ -365,7 +366,14 @@ private void putIntoVar(final JMeterVariables jmvars, final String name,
      * @return number of rows in resultSet
      * @throws SQLException
      */
-    private static int countRows(ResultSet resultSet) throws SQLException {
+    private int countRows(ResultSet resultSet) throws SQLException {
```

Comment:
I found out that with hsqldb (which is used in JDBC_TESTS.jmx), it is impossible to reach the countRows() function at all.
hsqldb's CallableStatement doesn't support output values of type ResultSet. And that is one precondition to reach this code.
hsqldb does support multiple result sets though, but not as output parameters.
So what do you think, is there more than hsqldb we run this code against? Are there any tests I am not aware of that test CallableStatements with output values of type ResultSet?

For your reference, see similar discussion here:
https://sourceforge.net/p/hsqldb/discussion/73674/thread/ec2be7ce/?limit=25

Reply:
how do we proceed here? that is the last open question before you can merge.
thanks!

Reply:
It seems you're right.
For now, no alternative idea except trying another in memory db that would have this feature.
I understand that you cover the other modifications in JDBC_TESTS.jmx right ?

Reply:
yes